### ISSUE TITLE
Add a rule to flag places where `ifBlank` and `ifEmpty` can be used

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#2840]
Kotlin provides the functions `ifBlank` and `ifEmpty` so we should flag the points where this functions could be used. Example:

`if (api.name.isBlank()) "John" else api.name`
can be replaced to:
`api.name.ifBlank { "John" }`

The same for `ifEmpty`.

## Context
https://twitter.com/kotlin/status/1277897234043097090

**Comment by schalkms** [comment]
This would go into the `style` ruleset in my point of view.

**Comment by mehwhatever** [comment]
If no one is working on this, would like to take a stab at it. :)

**Comment by schalkms** [comment]
Thanks @mehwhatever for the help! We really appreciate your help.
It's all yours. 🙂

**Comment by mehwhatever** [comment]
Do we also want to handle things like,

```
if (x.isBlank()) {
       return "A"
} else {
      return x
}
```

Though by opening that door, it would also make sense to catch these,
```
if (x.isBlank()) {
       return "A"
} 
return x
```

Which seems to be slightly tricky to catch.

On another note do we have a style rule for this,
```
if(something){
   return x
}
else {
   return y
}
```
to ```return if(something) x else y ```

**Comment by schalkms** [comment]
I think it's better to start simple and improve in subsequent PRs.

**Comment by BraisGabin** [comment]
> ```
> if (x.isBlank()) {
>        return "A"
> } 
> return x
> ```
I think that this case can be generalized and create a new rule for it. I think that the desired code here should be:
```
return if (x.isBlank()) {
       "A"
} else {
       x
}
```

I mean. a rule that detects an if that ends with a return and followed by that a return that could be changed by an `else`. Do you agree? But I think that we need a different issue to track this.

> On another note do we have a style rule for this,
> 
> ```
> if(something){
>    return x
> }
> else {
>    return y
> }
> ```

This should be handled by `formatting`. This is implemented by `ktlint` already.

**Comment by BraisGabin** [comment]
@mehwhatever I'm unassigning you from this issue. If you are still interested in this issue you can open the PR anyway :)
